### PR TITLE
Delivery page

### PR DESCRIPTION
**Pull request by nexussays**

### CODE REVIEW

**Comment by adamdehaven** [comment]
@nexussays I noticed on the /deliveries page, everyt time you scroll down on the page, the page automatically scrolls you back to the top. Reproduced when this was pushed to `next`; unsure if it happens locally.

File: src/components/app-root/app-root.tsx [hunk 1]
```diff
@@ -92,7 +87,7 @@ export class AppRoot {
               <stencil-route url="/activity" component="page-activity" />
```
Comment (mojowen) [inline src/components/app-root/app-root.tsx:87]: let's link the activities pages to their delivery locations

File: src/components/ui-geo-map/ui-geo-map.tsx [hunk 2]
```diff
@@ -13,22 +13,35 @@ export class UiGeoMap {
...
-      zoom: 12,
+      zoom: 16,
       fullscreenControl: false,
       mapTypeControl: false,
       streetViewControl: false,
+      scrollwheel: false,
+    });
+
+    this.marker = new google.maps.Marker({
+      icon: "/images/pin-pizza@1x.png",
```
Comment (mojowen) [inline src/components/ui-geo-map/ui-geo-map.tsx:33]: let's make this not transparent 

image

File: src/components/app-root/app-root.tsx (hunk 1 above)
Comment (mojowen) [reply src/components/app-root/app-root.tsx:87]: or I guess we could just remove this page and redirect `/activity` to `/deliveries`

File: src/components.d.ts [hunk 3]
```diff
@@ -42,23 +43,30 @@ export namespace Components {
...
+        "scrollId"?: string;
     }
     interface UiDynamicText {
         "format"?: (value: any) => string;
         "value": any | undefined;
     }
     interface UiGeoMap {
         "getCenter": () => Promise<google.maps.LatLng | undefined>;
-        "setCenter": (newCenter: google.maps.LatLngLiteral) => Promise<void>;
+        "setCenter": (newCenter: google.maps.LatLngLiteral, showMarker?: boolean) => Promise<void>;
```
Comment (mojowen) [inline src/components.d.ts:69]: map doesn't seem to update

image

File: src/components/ui-geo-map/ui-geo-map.tsx (hunk 2 above)
Comment (adamdehaven) [reply src/components/ui-geo-map/ui-geo-map.tsx:33]: Should be able to add a white background behind the transparency -- Sarah should be able to export a new asset.

File: src/components/app-root/app-root.tsx (hunk 1 above)
Comment (nexussays) [reply src/components/app-root/app-root.tsx:87]: I linked to the activity page from the "view more" link in the previous deliveries card. Not doing any location filtering though. Thoughts?

File: src/components.d.ts (hunk 3 above)
Comment (nexussays) [reply src/components.d.ts:69]: I'll take a look, it should work if the address is correct, but there is a similar issue of what to do when the address returns 404.

**Comment by mojowen** [comment]
@nexussays made some small updates to the trucks API in https://github.com/pizza-to-the-polls/pizzabase/pull/27

- Returns `region` instead of `identifier` (I don't think you're using this)
- Returns `isActive` on a truck
- `/trucks` endpoint can accept `location=<id>` to show for a location and `past=true` to toggle active vs all

File: src/components/ui-address-input/ui-address-input.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+const GMAPS_COMPONENT_MAPPING: { [key: string]: string } = {
+  sublocality: "city",
+  locality: "city",
+  postal_code: "zip",
+  route: "street",
+  street_number: "num",
+  administrative_area_level_1: "state",
+};
+
+const toFullAddress = (addressComponents: Array<{ short_name: string; types: Array<string> }>) => {
```
Comment (mojowen) [inline src/components/ui-address-input/ui-address-input.tsx:12]: @nexussays updated to match how pizzabase format's google addresses 

https://github.com/pizza-to-the-polls/pizzabase/blob/4f44dcec6e34a194cca8a96f15f00e4e3b2cd915/src/lib/validator/geocode.ts#L29-L60

**Review by mojowen — APPROVED** [review]

**Pull request merged**